x/interchainstaking/keeper/keeper.go

```diff
@@ -523,7 +523,23 @@ type RebalanceTarget struct {
 	Target string
 }
 
-func DetermineAllocationsForRebalancing(currentAllocations map[string]math.Int, currentSum math.Int, targetAllocations types.ValidatorIntents, existingRedelegations []types.RedelegationRecord) []RebalanceTarget {
+type RebalanceTargets []RebalanceTarget
+
+func (rb RebalanceTargets) RemoveDuplicates() RebalanceTargets {
+	encountered := make(map[string]bool)
+	result := make(RebalanceTargets, 0)
+
+	for _, r := range rb {
+		key := fmt.Sprintf("%v-%s-%s", r.Amount.String(), r.Source, r.Target)
+		if !encountered[key] {
+			encountered[key] = true
+			result = append(result, r)
+		}
+	}
+	return result
+}
+
+func DetermineAllocationsForRebalancing(currentAllocations map[string]math.Int, currentSum math.Int, targetAllocations types.ValidatorIntents, existingRedelegations []types.RedelegationRecord) RebalanceTargets {
```

Comment:
Should this function just always return the rebalance targets with the duplicates removed? 

@joe-bowman @ajansari95

Reply:
yes

Reply:
Yes, the logic is completely changed in v1.4, so this is just a workaround to make do until then.